src/update_agent/actor.rs

```diff
@@ -91,31 +139,50 @@ impl Handler<RefreshTick> for UpdateAgent {
                     "scheduling next agent refresh in {} seconds",
                     pause.as_secs()
                 );
-                Self::tick_later(ctx, pause);
+                actor.tick_later(ctx, pause);
             } else {
                 let update_timestamp = chrono::Utc::now();
                 actor.state_changed = update_timestamp;
-                Self::tick_now(ctx);
+                actor.tick_now(ctx);
             }
             actix::fut::ready(())
         });
 
         // Process state machine refresh ticks sequentially.
         ctx.wait(update_machine);
 
-        Box::pin(actix::fut::ok(()))
+        Box::pin(actix::fut::ok(self.state.clone()))
     }
 }
 
 impl UpdateAgent {
+    /// Cancel the scheduled refresh tick whose handle is stored in the update agent's
+    /// `tick_later_handle` field, if any.
+    fn cancel_scheduled_ticks(&mut self, ctx: &mut Context<Self>) {
```

Comment:
When transitioning "naturally" (no messages received from the D-Bus actor) from a "steady state" (e.g. `NoNewUpdates` or `UpdateAvailable`) to a new state, `tick_later_handle` is a handle to
a scheduled msg that has already been sent, so there is no need to cancel it. But it doesn't look like there is any harm in calling `ctx.cancel_future()` on a handle that is already sent, so for simplicity, always attempt to cancel.
But let me know if this doesn't make sense.